package.json

```diff
@@ -13,25 +14,38 @@
     "broccoli-plugin"
   ],
   "files": [
-    "index.js"
+    "index.js",
+    "index.ts",
+    "index.d.ts"
   ],
   "scripts": {
-    "lint": "eslint **/*.js",
-    "pretest": "multidep test/multidep.json",
-    "test": "mocha"
+    "build": "tsc -b .",
+    "clean": "tsc -b --clean .",
+    "lint": "eslint **/*.ts",
+    "test": "yarn build && yarn test:pre && yarn test:js",
+    "test:pre": "multidep test/multidep.json",
+    "test:js": "mocha -r ts-node/register **/*.test.ts",
```

Comment:
I have run into strange inconsistencies between `ts-node/register` and the regular `tsc -b`

To avoid differences between test and what is deployed, I have settled on simply building `tests` as well, then running mocha without ts-node: https://github.com/stefanpenner/resolve-package-path/blob/master/package.json#L31

Reply:
Done